simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileLargeDataSets.java

```diff
@@ -78,22 +82,27 @@ public BlockItem getNextBlockItem() throws IOException, BlockSimulatorParsingExc
     public Block getNextBlock() throws IOException, BlockSimulatorParsingException {
         currentBlockIndex++;
 
-        String nextBlockFileName = String.format(formatString, currentBlockIndex);
-        File blockFile = new File(blockstreamPath, nextBlockFileName);
-
-        if (!blockFile.exists()) {
+        final String nextBlockFileName = String.format(formatString, currentBlockIndex);
+        final Path localBlockStreamPath = Path.of(blockstreamPath).resolve(nextBlockFileName);
+        if (!Files.exists(localBlockStreamPath)) {
             return null;
         }
-
         try {
-            byte[] blockBytes = readFileBytes(blockFile.toPath());
+            final byte[] blockBytes =
+                    FileUtilities.readFileBytesUnsafe(localBlockStreamPath, RECORD_EXTENSION, GZ_EXTENSION);
+
+            if (Objects.isNull(blockBytes)) {
+                throw new NullPointerException(
+                        "Unable to read block file [%s]! Most likely not found with the extensions '%s' or '%s'"
+                                .formatted(localBlockStreamPath, RECORD_EXTENSION, GZ_EXTENSION));
+            }
 
-            LOGGER.log(INFO, "Loading block: " + blockFile.getName());
+            LOGGER.log(INFO, "Loading block: " + localBlockStreamPath.getFileName());
 
-            Block block = Block.PROTOBUF.parse(Bytes.wrap(blockBytes));
+            final Block block = Block.PROTOBUF.parse(Bytes.wrap(blockBytes));
             LOGGER.log(INFO, "block loaded with items size= " + block.items().size());
             return block;
-        } catch (ParseException e) {
+        } catch (final ParseException e) {
             throw new BlockSimulatorParsingException(e.getMessage());
```

Comment:
Quick note (that can be resolved in a later PR):
Never pass a message to a chained exception, always pass the whole exception.
If our custom exception does not have a chained exception constructor, add it.

Reply:
Noted, this was from the past, I will address it in #272.

Reply:
✅ Done.